internal/renderers/femtovg/lib.rs

```diff
@@ -29,36 +33,55 @@ mod fonts;
 mod images;
 mod itemrenderer;
 
+/// Trait that the FemtoVGRenderer uses to ensure that the OpenGL context is current, before running
+/// OpenGL commands. The trait also provides access to the symbols of the OpenGL implementation.
+pub trait OpenGLContextWrapper {
+    /// Ensures that the GL context is current.
+    fn ensure_current(&self) -> Result<(), PlatformError>;
+    fn swap_buffers(&self) -> Result<(), PlatformError>;
+    fn resize(&self, size: PhysicalWindowSize) -> Result<(), PlatformError>;
+    #[cfg(not(target_arch = "wasm32"))]
+    fn get_proc_address(&self, name: &std::ffi::CStr) -> *const std::ffi::c_void;
```

Comment:
This trait can't be made safe, can it?  What if this return a dangling pointer?

I believe the trait should be made unsafe. Or the get_proc_address should be made another way.
But the problem existed before  `new` should have been made unsafe

Reply:
Good question. I suppose it's up to the caller anyway to use `unsafe{}` before they can dereference the returned pointer?

In glutin this is not marked unsafe either (https://docs.rs/glutin/0.30.8/glutin/display/trait.GlDisplay.html#tymethod.get_proc_address ) that's the code the our impl is calling.

Reply:
That glutin trait is sealed. So it is sound. 


> I suppose it's up to the caller anyway to use unsafe{} before they can dereference the returned pointer?

But the caller can't possibly know the invarient:

```rust
// SAFETY: get_proc_address return a valid pointer for this because the OpenGLContectWrapper invariant states that the return pointer of the get_proc_address should be valid in this case
unsafe { do_something_with(context.get_proc_address()) }
```

However, in order to write this comment, there should be a safety invariant in place.
And I can just do 

```rust
impl OpenGLContextWrapper for MyStruct {
  fn get_proc_address(&self, name: &std::ffi::CStr) -> *const std::ffi::c_void { 
         0x123456 as *const _
  }
  // ...
}
```

And we get undefined behavior with the API, no `unsafe` needed